src/main/java/thePackmaster/powers/pixiepack/HorizonboundPower.java

```diff
@@ -0,0 +1,90 @@
+package thePackmaster.powers.pixiepack;
+
+import com.megacrit.cardcrawl.actions.common.DrawCardAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.localization.PowerStrings;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import com.megacrit.cardcrawl.powers.AbstractPower;
+import thePackmaster.SpireAnniversary5Mod;
+import thePackmaster.packs.PixiePack;
+import thePackmaster.powers.AbstractPackmasterPower;
+
+import java.util.ArrayList;
+
+public class HorizonboundPower extends AbstractPackmasterPower {
+    public static final String POWER_ID = SpireAnniversary5Mod.makeID("HorizonboundPower");
+    private static final PowerStrings powerStrings = CardCrawlGame.languagePack.getPowerStrings(POWER_ID);
+    public static final String NAME = powerStrings.NAME;
+    public static final String[] DESCRIPTIONS = powerStrings.DESCRIPTIONS;
+
+    ArrayList<AbstractCard> alreadyApplied = new ArrayList<>();
+
+    public HorizonboundPower(AbstractCreature owner, int amount) {
+        super(POWER_ID, NAME, PowerType.BUFF,false, owner, amount);
+    }
+
+    @Override
+    public void onInitialApplication() {
+        super.onInitialApplication();
+        amount = 0;
+        for (AbstractCard C:AbstractDungeon.actionManager.cardsPlayedThisTurn)
+        {
+            if (PixiePack.isForeign(C)) amount++;
+            for (AbstractCard K:AbstractDungeon.player.hand.group) {
+                if (PixiePack.isForeign(K))
```

Comment:
Have you tested what the logic in this if does to X-cost and unplayable cards? Might do nothing, might do something weird, best to avoid cards like those with any cost reduction effect.

X-cost cards have `cost == -1`, unplayable cards have `cost == -2`, you should probably just exclude those from this logic. In other words, change this if to `if (PixiePack.isForeign(K) && k.cost >= 0)` to exclude those.

You'll want to do the same for the other places you apply this cost reduction in onPlayCard and update too (or make a helper method to put this logic in).